basics/list-comprehension.md

```diff
@@ -0,0 +1,77 @@
+# List Comprehension
+
+
+## Definition
+List comprehensions creates a new list from an existing list that is a shorter syntax than normal. It also processes the list much faster than using a for loop.
+
+
+## Without List Comprehension
+```python
+animals = ["dog", "cat", "giraffe", "donkey", "ape"]
+arr = []
+
+for i in animals:
+  if len(i) == 3:
+    arr.append(i)
+    
+print(arr)
+```
+```python
+['dog', 'cat', 'ape']
```

Comment:
It would be good to show the output right at the print statement, so it is clear that it comes from `print(arr)`. It matters more when there's multiple prints, but I think it helps here too.

```suggestion
print(arr)  # Prints: ['dog', 'cat', 'ape']
```